instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/SnsCamelTest.groovy

```diff
@@ -0,0 +1,696 @@
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(4, 1) {
+
+        span(0) {
+          name "SNS.Subscribe"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "Subscribe"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(5, 1) {
+        span(0) {
+          name "SQS.ReceiveMessage"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "ReceiveMessage"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
+        }
+      }
+      trace(6, 3) {
+        span(0) {
+          name "SNS.Publish"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "Publish"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
+        }
+        span(1) {
+          name "SQS.ReceiveMessage"
+          kind CONSUMER
+          childOf span(0)
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "ReceiveMessage"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "http.user_agent" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
+        }
+        span(2) {
+          name "snsCamelTest"
+          kind INTERNAL
+          childOf span(0)
+          attributes {
+            "apache-camel.uri" "aws-sqs://${queueName}?amazonSQSClient=%23sqsClient"
+            "messaging.destination" queueName
+            "messaging.message_id" String
+          }
+        }
+      }
+      trace(7, 1) {
+        span(0) {
+          name "SQS.ReceiveMessage"
+          kind CLIENT
```

Comment:
Why is it CLIENT here and CONSUMER above?

Reply:
This is offspring of the modelling done for AWS SDK (as far as I remember). 
TLDR; each SQS response can have multiple messages, so each call (pooling) to SQS is modelled as a (HTTP) client, while each message consumed (if any) is a consumer span. Also, consumer spans are linked with producer spans as a single trace, while each SQS request is on its own.